File: py/ccwg-db-client.py
#!/usr/bin/env python3
#
# ccwg-db-client.py — Test client for ccwg-db-server.py (Python, stdlib only)
#
# RUN:
#   python3 ccwg-db-client.py
#
# REQUIRES:
#   Python 3 (uses only standard library: urllib.request, json, sys)
#
# DESCRIPTION:
#   Sends a POST request to the local ccwg-db-server with the hostnames
#   listed below (one per line) and prints the JSON response with pretty
#   formatting. Start ccwg-db-server.py first in another terminal.
#

import json
import sys
import urllib.request
import urllib.error


# --- Hostnames to look up (one per line) ---

HOSTNAMES = "google.com\nreddit.com"


# --- Server endpoint ---

SERVER_URL = "http://127.0.0.1:8999/"


# --- Send request ---

req = urllib.request.Request(
    SERVER_URL,
    data=HOSTNAMES.encode("utf-8"),
    method="POST",
)

try:
    resp = urllib.request.urlopen(req, timeout=30)
    body = resp.read().decode("utf-8")
except urllib.error.HTTPError as e:
    print("HTTP %d" % e.code, file=sys.stderr)
    body = e.read().decode("utf-8")
except urllib.error.URLError as e:
    print("Request failed: %s" % e.reason, file=sys.stderr)
    sys.exit(1)


# --- Pretty-print JSON response ---

try:
    data = json.loads(body)
    print(json.dumps(data, indent=2))
except ValueError:
    print(body)

File: php/ccwg-db-client.php
<?php
declare(strict_types=1);

/*
 * ccwg-db-client.php — Test client for ccwg-db-server.php (PHP + curl)
 *
 * RUN:
 *   php ccwg-db-client.php
 *
 * REQUIRES:
 *   PHP with the curl extension (enabled by default in most distributions)
 *
 * DESCRIPTION:
 *   Sends a POST request to the local ccwg-db-server with the hostnames
 *   listed below (one per line) and prints the JSON response with pretty
 *   formatting. Start ccwg-db-server.php first in another terminal.
 */


// --- Hostnames to look up (one per line) ---

$hostnames = "google.com\nreddit.com";


// --- Server endpoint ---

$server_url = 'http://127.0.0.1:8999/';


// --- Send request ---

$ch = curl_init($server_url);
curl_setopt($ch, CURLOPT_POST, true);
curl_setopt($ch, CURLOPT_POSTFIELDS, $hostnames);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_TIMEOUT, 30);
curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 5);

$response = curl_exec($ch);

if ($response === false) {
	fprintf(STDERR, "curl error: %s\n", curl_error($ch));
	curl_close($ch);
	exit(1);
}

$http_code = (int)curl_getinfo($ch, CURLINFO_HTTP_CODE);
curl_close($ch);

if ($http_code !== 200) {
	fprintf(STDERR, "HTTP %d\n", $http_code);
}


// --- Pretty-print JSON response ---

$json = json_decode($response, true);

if ($json === null) {
	echo $response . "\n";
} else {
	echo json_encode($json, JSON_PRETTY_PRINT) . "\n";
}
